addons/hr_work_entry_contract/models/hr_work_entry.py

```diff
@@ -54,6 +55,34 @@ def _init_column(self, column_name):
                 WHERE _hwe.id = ANY(result.entry_ids)
             """)
 
+    def _get_missing_work_entry(self):
+        data = {}
+        work_entries_dict = {}
+        for work_entry in self:
+            key = (work_entry.date_start.date(), work_entry.employee_id)
+            if key not in work_entries_dict:
+                work_entries_dict[key] = []
+            work_entries_dict[key].append(work_entry)
+
+        for key, work_entries in work_entries_dict.items():
+            total_duration = sum(work_entry.duration for work_entry in work_entries)
+            if total_duration < work_entries[0].contract_id.resource_calendar_id.hours_per_day:
+                for work_entry in work_entries:
+                    data[work_entry] = total_duration
+        return data
```

Comment:
Why not just return a recordset? I see that for every call to this function, you only use the keys in the end, which is forcing you to do loops to call methods working in batch...
You could have a recordset for the data like this
```py
data = self.env['hr.work.entry']
```
```suggestion
                data |= work_entries
        return data
```